lldb/source/Utility/ArchSpec.cpp

```diff
@@ -1426,7 +1426,8 @@ bool ArchSpec::IsFullySpecifiedTriple() const {
     if ((user_specified_triple.getVendor() != llvm::Triple::UnknownVendor) ||
         TripleVendorWasSpecified()) {
       const unsigned unspecified = 0;
-      if (user_specified_triple.getOSMajorVersion() != unspecified) {
+      if (!user_specified_triple.isOSDarwin() ||
+          user_specified_triple.getOSMajorVersion() != unspecified) {
```

Comment:
Could you please make this change upstream with a unit test? ArchSpec is super sensitive and I don't want any divergencies from llvm.org if we can at all avoid it.

Reply:
I'll sure, I can do a change upstream as well (but Id like to get this in here first to get the unified layout unblocked).